src/components/FacetInsights.js

```diff
@@ -124,14 +126,32 @@ export default class FacetInsights extends React.Component<FacetInsightsDefaultP
     return 'list';
   }
 
-  facetForField(fieldName: string, facetMap: Map<string, SearchFacet>) {
+  /**
+   * Filter Buckets in Facet that are already applied to the search filters.
+   * Buckets of type tagcloud and timeseries do not need to be filtered.
+   */
+  filterFacetBuckets = (facet: SearchFacet, type: string, facetFiltersMap: Map<string, FacetFilter>): SearchFacet => {
+    if (type === 'tagcloud' || type === 'timeseries') {
+      return facet;
+    }
+    const filteredBuckets = [];
```

Comment:
Instead of `forEach`, [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) works well for this.
```suggestion
    const filteredBuckets = facet.buckets.reduce((accumBuckets, currentBucket) => {
       return !facetFiltersMap.get(bucket.filter)
         ? [...accumBuckets, currentBucket]
         : accumBuckets; 
    }, []);
```